engine/baml-runtime/src/cli/serve/mod.rs

```diff
@@ -340,7 +330,29 @@ Tip: test that the server is up using `curl http://localhost:{}/_debug/ping`
             self.port,
         );
 
-        service.await?;
+        // Add graceful shutdown with Ctrl+C handling, and a clean message for exit to the client
+        let shutdown = async {
+            match tokio::signal::ctrl_c().await {
+                Ok(()) => {
+                    // Print message and ensure it's flushed
+                    println!("\nExiting BAML...");
+                    std::io::stdout().flush().unwrap_or_default();
+                    // Small delay to ensure message is visible
+                    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+                }
+                Err(err) => {
+                    log::error!("Error handling Ctrl+C: {}", err);
+                }
+            }
+        };
+
+        // Set ctrl-c handler to ignore so Python doesn't intercept it, otherwise python doesnt like it
```

Comment:
As discussed in Discord, let's try moving this into [`invoke_runtime_cli`](https://github.com/BoundaryML/baml/blob/canary/engine/language_client_python/src/lib.rs#L12-L13)

Everything in `baml-runtime` is shared across all 4 builds of BAML (Python, Node, Ruby, and the internal-only Rust CLI) so I'd rather not muck with default signal handling behavior unless we have to - e.g. right now it's really easy for someone to misunderstand how the current code works, because it's easy to read it as setting up a tokio ctrl-c handler and then uninstalling all signal handlers (even though it's actually the other way around due to how Rust async works)